src/common/arrow/src/native/read/array/binary.rs

```diff
@@ -308,7 +309,18 @@ fn try_new_binary_array<O: Offset>(
     validity: Option<Bitmap>,
 ) -> Result<Box<dyn Array>> {
     if matches!(data_type, DataType::Utf8 | DataType::LargeUtf8) {
-        let array = Utf8Array::<O>::try_new(data_type, offsets, values, validity)?;
+        // todo!("new string")
+        let array =
+            Utf8Array::<O>::try_new(data_type, offsets, values, validity).map_err(|err| {
+                Error::External(
+                    "Encountered invalid utf8 data for string type, \
+                        if you were reading column with string type from a table, \
+                        it's recommended to alter the column type to `BINARY`.\n\
+                        Example: `ALTER TABLE <table> MODIFY COLUMN <column> BINARY;`"
```

Comment:
It is better to put this example into the upper level of the relevant function to return, and it is more appropriate to return only the utf8 error here.

Reply:
I'm afriad that we can not cover all callee of this function. I prefer to let it be here for a while during the transition period, and removing it until most related users has properly updated.